// @stellar/mpp/dist/env.js
export function parseRequired(name) {
    const value = process.env[name];
    if (value === undefined || value === '') {
        throw new Error(`${name} is required`);
    }
    return value;
}
export function parseOptional(name, fallback) {
    const value = process.env[name];
    if (value !== undefined && value !== '')
        return value;
    return fallback;
}
export function parsePort(name = 'PORT', fallback) {
    const raw = process.env[name];
    if (raw === undefined || raw === '') {
        if (fallback !== undefined)
            return fallback;
        throw new Error(`${name} is required`);
    }
    const port = Number(raw);
    if (!Number.isInteger(port) || port < 1 || port > 65535) {
        throw new Error(`Invalid ${name}: ${raw}. Must be an integer between 1 and 65535.`);
    }
    return port;
}
export function parseStellarPublicKey(name) {
    const value = parseRequired(name);
    if (!value.startsWith('G') || value.length !== 56) {
        throw new Error(`${name} must be a Stellar public key (G...)`);
    }
    return value;
}
export function parseStellarSecretKey(name) {
    const value = parseRequired(name);
    if (!value.startsWith('S') || value.length !== 56) {
        throw new Error(`${name} must be a Stellar secret key (S...)`);
    }
    return value;
}
export function parseContractAddress(name) {
    const value = parseRequired(name);
    if (!value.startsWith('C') || value.length !== 56) {
        throw new Error(`${name} must be a contract address (C...)`);
    }
    return value;
}
export function parseHexKey(name, length = 64) {
    const value = parseRequired(name);
    const hexRegex = new RegExp(`^[0-9a-fA-F]{${length}}$`);
    if (!hexRegex.test(value)) {
        throw new Error(`${name} must be ${length} hex characters`);
    }
    return value;
}
export function parseCommaSeparatedList(value) {
    return value
        .split(',')
        .map((s) => s.trim())
        .filter(Boolean);
}
export function parseNumber(name, opts) {
    const raw = process.env[name];
    if (raw === undefined || raw === '') {
        if (opts?.fallback !== undefined)
            return opts.fallback;
        throw new Error(`${name} is required`);
    }
    const num = Number(raw);
    if (isNaN(num)) {
        throw new Error(`Invalid ${name}: ${raw}. Must be a number.`);
    }
    if (opts?.min !== undefined && num < opts.min) {
        throw new Error(`${name} must be >= ${opts.min}`);
    }
    if (opts?.max !== undefined && num > opts.max) {
        throw new Error(`${name} must be <= ${opts.max}`);
    }
    return num;
}
